R/PythonPkgWrapperUtils.R

```diff
@@ -0,0 +1,1106 @@
+#' * `functionFilter` takes as input the metadata for a generated function and either modifies it
+#'   or returns NULL to omit it from the set of generated functions. The metadata object is a list
+#'   having fields:
+#'   ```
+#'   'name': character
+#'   'args': named list having fields:
+#'       'args': a list of the argument names
+#'       'varargs':  character
+#'       'keywords': character
+#'       'defaults': character
+#'   'doc': character
+#'   'module':character
+#'   ```
+#'   Please see [inspect.getargspec](https://docs.python.org/2/library/inspect.html#inspect.getargspec)
+#'     for more information about the named list `args`.
+#'   See example 2.
+#'
+#' * `classFilter` takes as input the metadata for a generated class and either modifies it
+#'   or returns NULL to omit it from the set of generated classes The metadata object is a list
+#'   having fields:
+#'   ```
+#'   'name': character
+#'   'constructorArgs': named list having fields:
+#'       'args': a list of the argument names
+#'       'varargs':  character
+#'       'keywords': character
+#'       'defaults': character
+#'   'doc': character
+#'   'methods':named list having fields:
+#'       'name': character
+#'       'doc': character
+#'       'args': named list having fields:
+#'           'args': a list of the argument names
+#'           'varargs':  character
+#'           'keywords': character
+#'           'defaults': character
+#'   ```
+#'   Please see [inspect.getargspec](https://docs.python.org/2/library/inspect.html#inspect.getargspec)
+#'     for more information about the named list `args`.
+#'   See example 3.
+#'
+#' * `transformReturnObject` is used to intercept and modify the values returned by the
+#'   auto-generated R functions.`transformReturnObject` will be applied to the returned values
+#'   from all generated functions. The transformation cannot depend on the function which generated
+#'   the returned value. See example 5.
+#'
+#' @note
+#' * `generateRWrappers` should be called at load time.
+#' * `generateRWrappers` and `generateRdFiles` must be called with corresponding parameters to ensure
+#'    all R wrappers has sufficient documentation.
+#' @examples
+#' # 1. Generate R wrappers for all functions, classes, and enums in "pyPackageName.aModuleInPyPackageName"
+#'
+#' callback <- function(name, def) {
+#'   setGeneric(name, def)
+#' }
+# .NAMESPACE <- environment()
+#' assignEnumCallback <- function(name, keys, values) {
+#'   assign(name, setNames(values, keys), .NAMESPACE)
+#' }
+#' PythonEmbedInR::generateRWrappers(
+#'   pyPkg = "pyPackageName",
+#'   container = "pyPackageName.aModuleInPyPackageName",
+#'   setGenericCallback = callback,
+#'   assignEnumCallback = assignEnumCallback)
+#'
+#' # 2. Generate R wrappers for module "pyPackageName.aModuleInPyPackageName", omitting function "myFun"
+#'
+#' myfunctionFilter <- function(x) {
+#'   if (any(x$name == "myFun")) NULL else x
+#' }
+#' PythonEmbedInR::generateRWrappers(
```

Comment:
Remove reference to the package `PythonEmbedInR`.

Reply:
Nice catch. These have been removed.